shell/common/shell.cc

```diff
@@ -876,6 +877,81 @@ void Shell::UpdateIsolateDescription(const std::string isolate_name,
   vm_->GetServiceProtocol()->SetHandlerDescription(this, description);
 }
 
+void Shell::SetNeedsReportTimings(bool value) {
+  needs_report_timings_ = value;
+}
+
+void Shell::ReportTimings() {
+  FML_DCHECK(is_setup_);
+  FML_DCHECK(task_runners_.GetGPUTaskRunner()->RunsTasksOnCurrentThread());
+
+  // The engine might have been destructed in ~Shell while the rasterizer is
+  // still alive and calling this from OnFrameRasterized.
+  if (!engine_) {
```

Comment:
This condition is not possible as the destruction of the shell will only happen after the engine is destroyed.

Reply:
Removed this racy access to `unique_ptr<Engine> engine_` on the GPU thread, and modified `GetEngine` as we discussed.

Reply:
Thanks. Updates look safe.

Reply:
Is this the renewed LGTM? 😄

Reply:
Oh. Sorry. Yeah :)